src/app.js

```diff
@@ -219,6 +228,36 @@ gpii.app.handleSessionStop = function (that, keyedOutUserToken) {
     }
 };
 
+/**
+ * Listen on uncaught exceptions and display it to the user is if it's interesting.
+ * @param that {Component} An instance of gpii.app.
+ */
+gpii.app.handleError = function (that) {
+    fluid.onUncaughtException.addListener(function (err) {
```

Comment:
This is a pretty bulky use of nested code, as well as hiding some state in a closure. I think this would be a lot clearer if it was all shifted out to top level, as well as storing its state on a component somewhere - e.g. on the app itself.
I suggest eliminating the handleError function and listener, and instead simply writing something like

fluid.onUncaughtException.addListener(function (err) {
    var app = fluid.queryIoCSelector(fluid.rootComponent, "gpii.app");
    if (app.length > 0) {
        app[0].handleUncaughtException(err);
    }, "gpii.app", "last");